firmware/application/ui_navigation.cpp

```diff
@@ -555,6 +560,76 @@ void SystemStatusView::rtc_battery_workaround() {
     }
 }
 
+void SystemStatusView::new_sdcard_structure_checker() {
+    const std::filesystem::path root_dir = u"/";
+    const std::filesystem::path resources_dir = u"SYS";
+    std::vector<std::filesystem::path> directories = scan_root_directories(root_dir);
+
+    auto scan_result = std::find(directories.begin(), directories.end(), resources_dir);
+
+    auto sd_status = sd_card::status();
+
+    if (
+        nav_.is_valid() &&
+        (scan_result == directories.end()) &&
+        (sd_status == sd_card::Status::Mounted)) {
+        nav_.display_modal(
+            "Warning",
+            "        WARNING!!!\n"
+            "Sdcard content changed\n"
+            "in this update,\n"
+            "Press YES to auto move.\n"
+            "After finished,\n"
+            "you should move your own fil-\n"
```

Comment:
- "Sdcard" should be "SD Card"
- I don't think the word "files" should be split across lines.
- "original folder" probably should be "original folders" or "root folders"
- "which is now" probably should be "which are now"